src/app/profile/profile.component.html

```diff
@@ -0,0 +1,37 @@
+<form [formGroup]="profileForm" (ngSubmit)="onSubmit(profileForm.value)">
+    <h2>Profile</h2>
+    <p>
+        <label for="firstName">First Name</label>
+        <input type="text" id="firstName" formControlName="firstName">
+        <span class="error-message" *ngIf="firstName.hasError('required') && firstName.dirty">Required Field</span>
```

Comment:
QUESTION - should logic be here, or should this have been a function like i did for email validation. I didnt want to clutter the component so that is why I put it here.

Reply:
I personally have a preference to put a well named function in the component and keeping the template lighter.